app/js/components/orderEntry/OrdersTable.jsx

```diff
@@ -9,21 +9,86 @@ import DrugOrderDetails from './DrugOrderDetails';
 import fetchOrders from '../../actions/fetchOrders';
 import { DRUG_ORDER } from './orderTypes';
 import { setSelectedOrder } from '../../actions/orderAction';
+import createLabOrder from '../../actions/createLabOrder';
+import { successToast, errorToast } from '../../utils/toast';
+import toggleUrgency from '../../actions/labOrders/toggleUrgency';
 
 export class OrdersTable extends PureComponent {
+  state = {
+    recentUrgency: '',
+    previousOrderId: '',
+  };
+
   componentDidMount() {
     this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
   }
 
-  componentDidUpdate(prevProps, prevState) {
+  componentDidUpdate(prevProps) {
+    const { previousOrderId, recentUrgency } = this.state;
+    const {
+      status: { added, error },
+      errorMessage,
+      labOrderData,
+    } = this.props.createLabOrderReducer;
     if (this.props.patient.uuid !== prevProps.patient.uuid) {
       this.props.dispatch(fetchOrders(null, this.props.patient.uuid));
     }
+    if (added && labOrderData !== prevProps.createLabOrderReducer.labOrderData) {
+      this.props.dispatch(toggleUrgency(
+        previousOrderId,
+        labOrderData.orders[0].uuid,
+        recentUrgency,
+      ));
+      successToast('urgency successfully changed');
+    }
+    if (error) {
```

Comment:
I don't see any case where this errorToast would be displayed since the action dispatched won't be doing any server side calls so we're sure of no server error for one.

Reply:
I guess you did not look at the action creator then. the toggling urgency implemented here is not just to update the draft data but the previously created order, so, an api call is definitely done in the action creator..

Reply:
this is the action for `toggleUrgency`
```
const toggleUrgency = (previousOrderId, newOrderId, newUrgency) => ({
  type: TOGGLE_URGENCY,
  previousOrderId,
  newOrderId,
  newUrgency,
});
```
so how is an api call definitely done here ? I guess I'm not understanding

Reply:
please look at the toggleUrgency method, the last line is where the action that makes the api call is dispatched.